### DISCUSSION TITLE
64bit Home Assistant OS not supported?

### DISCUSSION THREAD

**Discussion by RudyReindeer** [OpenCCU/OpenCCU#2147] [🇺🇸 Q&A] [answered]
Do I assume correctly that a 32Bit addon will not run on a 64Bit OS as stated in "[Switch to 64bit kernel+OS environment (x86_64, aarch64)](https://github.com/jens-maus/RaspberryMatic/issues/903)"?
And that there is no translation layer?

Facts:
I have installed a new HomeAssistant (ver 9.4 with 64Bit) on a RASPI 4. 
So far all good. I have the Homematic running on the same Raspi (was running with homegear, so functional) and want to migrate this too. I can add the repository as described and the log will count 2 new "add-ons" as expected. The  GUI shows no change, i. e. no install-able addons as described in the documentation.
I did some research, and it seems that this is a 64Bit /32bit issue. The installation documentation does not state this and I want to verify my assumption before adding more steps. I am happy to add information / logs, if they provide additional information.
Rudy

**Comment by jens-maus** [comment, marked as answer]
No, RaspberryMatic runs on any 64bit platform even thought its main binaries for the homematic services are 32bit only. There is indeed a translation layer (multiarch) with 32bit support libraries in /usr/lib32, etc. The only requirement is that the 64bit CPU still comes with the necessary 32bit support instructions, which fortunately the CPUs on a RaspberryPi do. The only CPUs I currently know which do not support 32bit applications are the Apple Silicon ARM cpus because Apple unfortunately decided to drop the necessary 32bit instruction sets.

But for your environment everything should work as expected.

**Reply by RudyReindeer** [reply]
Thanks for the VERY fast answer. I tried this several times yesterday before posting this. Now the addon appears.
Case closed.
